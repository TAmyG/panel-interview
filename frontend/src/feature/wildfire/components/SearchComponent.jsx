import React, { useState } from 'react';
import { useForm } from '../../../hooks/useForm';
import { months } from '../data/months';

export const SearchComponent = ({ setMonthYear }) => {
    const [searchMonth, setSearchMonth] = useState({ value: 'JAN' });
    const { searchYear, onInputChange } = useForm({ searchYear: '2020' });

    const onSearchSubmit = (e) => {
        e.preventDefault();
        setMonthYear(searchMonth, searchYear);
    };

    const onSelectChange = (e) => {
        setSearchMonth(() => ({
            value: e.target.value,
        }));
    };

    return (
        <form aria-label="form" onSubmit={onSearchSubmit}>
            <div className="row mt-4">
                <div className="col">
                    <select
                        aria-label="month"
                        value={searchMonth.value}
                        className="form-control"
                        onChange={onSelectChange}
                    >
                        {months.map((month) => (
                            <option
                                data-testid="select-option"
                                key={month}
                                value={month}
                            >
                                {month}
                            </option>
                        ))}
                    </select>
                </div>
                <div className="col">
                    <input
                        aria-label="year"
                        type="number"
                        placeholder="Year"
                        className="form-control"
                        name="searchYear"
                        autoComplete="off"
                        value={searchYear}
                        onChange={onInputChange}
                        required={true}
                    />
                </div>
                <div className="col">
                    <button className="btn btn-outline-primary">Search</button>
                </div>
            </div>
        </form>
    );
};
