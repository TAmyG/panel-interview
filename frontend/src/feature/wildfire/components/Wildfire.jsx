import React from 'react';
import PropTypes from 'prop-types';

/**
 * {"title":"Wildfire - E of Mesa, Arizona - United States","geometry":[{"geoComponent":{"ISO3166Alpha3":"USA"}}]}
 * Component to render title and Country
 * @param {props with wildfire info from backend} wildfire
 * @returns react comnponent
 */
export const Wildfire = ({ title, index, geometry }) => {
    const country = geometry[0].geoComponent.ISO3166Alpha3;
    const { date } = geometry[0];
    return (
        <tr>
            <th scope="row">{index + 1}</th>
            <td>{title}</td>
            <td>{country}</td>
            <td>{date}</td>
        </tr>
    );
};

Wildfire.propTypes = {
    title: PropTypes.string.isRequired,
    index: PropTypes.number.isRequired,
    geometry: PropTypes.array.isRequired,
};
