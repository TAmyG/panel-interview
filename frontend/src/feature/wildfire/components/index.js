export * from './SearchComponent';
export * from './WildfiresEvents';