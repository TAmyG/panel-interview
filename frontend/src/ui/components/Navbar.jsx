import React from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import nasaLogo from '../../assets/nasa-logo.svg';

export const Navbar = () => {
    const user = {};

    const onLogout = () => {
        console.log('Logout and navigate to home');
    };
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2">
            <Link className="navbar-brand" to="/">
                <img src={nasaLogo} style={{ width: '50', height: '50' }} />
                NASA Wilfires
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">
                    <NavLink
                        className={({ isActive }) =>
                            `nav-item nav-link ${isActive ? 'active' : ''}`
                        }
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>
        </nav>
    );
};
