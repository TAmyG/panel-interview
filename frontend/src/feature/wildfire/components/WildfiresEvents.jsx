import React from 'react';
import PropTypes from 'prop-types';
import { useFetchWildfires } from '../hooks/useFetchWildfires';
import { Wildfire } from './Wildfire';

export const WildfiresEvents = ({ month, year }) => {
    const [wildfires, isLoading, ohNo] = useFetchWildfires(month, year);
    return (
        <>
            {isLoading && <h1>Loading...</h1>}

            {!isLoading &&
                (ohNo ? (
                    <h1>Oh No!</h1>
                ) : (
                    <table className="table table-hover">
                        <thead>
                            <tr>
                                <th scope="col">#</th>
                                <th scope="col">Name</th>
                                <th scope="col">Country</th>
                                <th scope="col">Date</th>
                            </tr>
                        </thead>
                        <tbody>
                            {wildfires.map((fire, index) => (
                                <Wildfire
                                    key={fire.id}
                                    index={index}
                                    {...fire}
                                />
                            ))}
                        </tbody>
                    </table>
                ))}
        </>
    );
};

Wildfire.propTypes = {
    month: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
};
