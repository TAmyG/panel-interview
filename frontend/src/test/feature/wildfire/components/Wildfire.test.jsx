import { render, screen } from '@testing-library/react';
import { expect, vi } from 'vitest';
import { Wildfire } from '../../../../feature/wildfire/components/Wildfire';

const props = {
    title: 'Fire event',
    index: 1,
    geometry: [
        {
            date: '2023-03-6',
            geoComponent: {
                ISO3166Alpha3: 'CAN',
            },
        },
    ],
};

describe('Testing on <Wildfire/>', () => {
    test('should match with the snapshot', () => {
        const result = render(<Wildfire {...props} />);
        expect(result).toMatchSnapshot();
    });

    test('should not render date value', async () => {
        const { geometry, ...newProps } = props;
        newProps.geometry = [
            {
                geoComponent: {
                    ISO3166Alpha3: 'CAN',
                },
            },
        ];
        render(<Wildfire {...newProps} />);
        const dateTd = await screen.findAllByLabelText('td');

        expect(dateTd[2].textContent).not.toBeTruthy();
    });
});
