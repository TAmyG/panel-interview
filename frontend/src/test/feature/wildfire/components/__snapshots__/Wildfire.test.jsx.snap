// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing on <Wildfire/> > should match with the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td
          aria-label="td"
        >
          Fire event
        </td>
        <td
          aria-label="td"
        >
          CAN
        </td>
        <td
          aria-label="td"
        >
          2023-03-6
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <th
        scope="row"
      >
        2
      </th>
      <td
        aria-label="td"
      >
        Fire event
      </td>
      <td
        aria-label="td"
      >
        CAN
      </td>
      <td
        aria-label="td"
      >
        2023-03-6
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
